api/transport/response.go

```diff
@@ -45,5 +45,5 @@ type ResponseWriter interface {
 	// SetApplicationError specifies that this response contains an
 	// application error. If called, this MUST be called before any invocation
 	// of Write().
-	SetApplicationError()
+	SetApplicationError(err error)
```

Comment:
This is a breaking change, and cannot be made. We have committed to this API for v1.